### PR TITLE
Barricade

### PR DESCRIPTION
**Pull request by TerryKang**

### CODE REVIEW

File: src/Barricade.cpp [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+		this->texture.setAlpha(30);
+}
+
+
+void Barricade::onCollision() {
+	// Do nothing for now
+}
+
+void Barricade::collidingProjectile(int damage) {
+	this->health = health - damage;
```
Comment (isaacmorneau) [inline src/Barricade.cpp:73]: should this not be `this->health -= damage;`?

File: src/Barricade.h [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+	void setAngle(double a);//sets angle of sprite to
+
+	double getAngle(); //returns sprites angle
+
+	bool isPlaceable();
+
+	bool checkPlaceablePosition(float, float, float, float, CollisionHandler*);
+
+	Barricade();
```
Comment (isaacmorneau) [inline src/Barricade.h:31]: The constructor and destructor should be the first two entities in a class.

Not yet in the style guide, I will add it.

File: src/Barricade.h [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	virtual ~Barricade();
+
+private:
```
Comment (isaacmorneau) [inline src/Barricade.h:34]: Defaults should be specified in the constructor not in the definitions of members. When implementing the constructor the with defaults it should be also using member initializers

File: src/GameManager.h [hunk 4]
```diff
@@ -54,7 +60,9 @@ class GameManager {
 	std::map<unsigned int, Marine*> marineManager;
 	std::map<unsigned int, Object*> objectManager;
 	std::map<unsigned int, Zombie*> zombieManager;
-    std::map<unsigned int, Turret*> turretManager;
+    	std::map<unsigned int, Turret*> turretManager;
```
Comment (isaacmorneau) [inline src/GameManager.h:63]: Tabs are not allowed, please use 4 spaces as per style guide

**Comment by jacobmcphail** [comment]
Think this PR is in the wrong repo...

**Comment by isaacmorneau** [comment]
That explains why it came out of the blue with only a couple changes, oh well. At least you know what would have been an issue. If it is tell me and ill close it.

**Pull request closed without merging**